proposals/p3797.md

```diff
@@ -0,0 +1,278 @@
+    it's likely we'll support syntax similar to `f#"..."` for interpolated
+    string literals. The `r#` syntax offers consistency with this, and will
+    hopefully be recognizable to users.
+-   Consistency with Rust.
+    -   Rust uses `r#"..."` for raw string literals, whereas Carbon uses
+        `#"..."`.
+-   Introduces another code execution path in lexing identifiers. This likely
+    causes a slowdown;
+    [PR #3044](https://github.com/carbon-language/carbon-lang/pull/3344)
+    indicates roughly 2%, although that was run on a system with noisy
+    benchmarks -- details would require a better system for benchmark. Note 2%
+    could represent that `r` is 1-in-55 identifiers with a 100% slowdown with
+    linear cost scaling for other similar code, or it could indicate that the
+    additional code path causes incremental slowdown but if other code (such as
+    `f#"..."`) used the same codepath it may instead have constant cost scaling
+    (negligible incremental cost). This may also be either reduced or become
+    more significant if we enable tail calls and other optimizations. As a
+    consequence, the precise overhead is difficult to quantify at this time.
+
+Various other prefixes have been discussed, mostly using a special character
+prefix in order to restrict the lexing impact. In particular:
+
+-   `\` prefix, as in `\identifier`.
+    -   Similar to `\` escaping in strings.
+    -   More intuitive "escaping" semantic for some developers versus `r#`.
+    -   Creates a different meaning for `\n` as an identifier versus `\n` as a
+        character escape.
+        -   Some of this could be addressed by restricting `\` raw identifiers
+            to only keywords in the language, meaning `\n` would only be a
+            character escape. The alternative
+            [Restrict raw identifier syntax to current and future keywords](#restrict-raw-identifier-syntax-to-current-and-future-keywords)
+            applies to this solution.
+-   `#` prefix without `r`, as in `#identifier`.
+    -   Would be more consistent with string literals, and avoid the lexing
+        overhead.
+    -   We are considering using a `#` prefix for metaprogramming, so the `r`
+        offers a way to keep the `#` prefix available for other purposes.
+    -   `#if` may look to C++ developers like a compiler directive, rather than
+        a raw identifier for `if`.
+-   `@` prefix, as in `@identifier`.
+    -   Consistent with C#.
+    -   We've also discussed using a `@` prefix for attributes, similar to
+        Python. Similar to `#`, this would be conflicting.
+-   <code>\`</code> wrapping, as in <code>\`identifier\`</code>.
+
+    -   Consistent with Swift.
+    -   We prefer not to use backticks for Carbon syntax so that it is easy to
+        write in Markdown, which uses backticks for inline code. For example,
+        the label for this alternative requires inline html:
```

Comment:
You can also write this without inline HTML as: ``` `` ` `` ``` (and I wrote that as ```` ``` `` ` `` ``` ````). And likewise ``` `` `identifier` `` ```.

Reply:
Fixed